src/Neo/Cryptography/Base58.cs

```diff
@@ -34,7 +34,7 @@ public static class Base58
         /// <returns>A byte array that is equivalent to <paramref name="input"/>.</returns>
         public static byte[] Base58CheckDecode(this string input)
         {
-            if (input is null) throw new ArgumentNullException(nameof(input));
+            if (string.IsNullOrEmpty(input)) throw new ArgumentException(nameof(input));
```

Comment:
Why change the exception?

Reply:
Because the `string input` parameter is always non-null.

Reply:
Its not `non-null`. Than you should be doing `string.IsNullOrWhitespace`